modules/ROOT/pages/_partials/sync/js-client-offline-and-conflict.adoc

```diff
@@ -1,46 +1,63 @@
 
-== Querying local cache
+Developers can attach listeners to get notification about if a update is applied on the server or not, and take appropriate actions.
 
-By default client will save all performed query results in the cache.
-Data will be available to be used when application goes offline.
-Queries are cached out of the box based on the type and `id` field.
-When performing mutations that affects some queries users can use `refetchQueries` or `update` methods:
+== Mutations and Local Cache
+
+By default queries are cached based on the type and `id` field, and the results of performed queries are cached as well and they will be available when the client is offline.
+
+Because of this, when mutations that can change query results are performed, the `refetchQueries` or `update` options of the `mutate` method should be used to ensure the local cache is kept up to date.
+
+In the following example, the app will perform an `ADD_TASK` mutation which will create a new task. The app also has a `GET_TASKS` query to list all the tasks. In order to make sure the cache for the `GET_TASKS` query is kept up to date whenever a new task is created, the `update` option is used to add the newly created task to the cache:
 
 ----
-    client.mutate<Task>({
-      mutation: ADD_TASK, variables: item,
-      optimisticResponse: createOptimisticResponse('createTask', 'Task', item),
-      update: this.updateCacheOnAdd
+  client.mutate<Task>({
+    mutation: ADD_TASK, variables: item,
+    update: this.updateCacheOnAdd
+  });
+
+  updateCacheOnAdd(cache, { data: { createTask } }) {
+    let { allTasks } = cache.readQuery({ query: GET_TASKS });
+    if (allTasks) {
+      if (!allTasks.find((task) => task.id === createTask.id)) {
+        allTasks.push(createTask);
+      }
+    } else {
+      allTasks = [createTask];
+    }
+    cache.writeQuery({
+      query: GET_TASKS,
+      data: {
+        'allTasks': allTasks
+      }
     });
+  }
 ----
 
-== Making modifications when offline
+For more information, please check out link:https://www.apollographql.com/docs/react/essentials/mutations.html#props[Apollo's document about mutations].
+
+// I don't really get what is section is for. Needs more context and code example
+// == Making Modifications When ffline
 
-AeroGear Sync SDK provides queue that stores mutations performed when offline.
-By default queue saves data in storage to be available after application restarts.
-Queue will hold requests until application will come back online.
+// AeroGear Sync SDK provides queue that stores mutations performed when offline.
+// By default queue saves data in storage to be available after application restarts.
+// Queue will hold requests until application will come back online.
 
-Developers can adjust how queue will process new mutations by supplying custom `NetworkStatus` implementation.
+// Developers can adjust how queue will process new mutations by supplying custom `NetworkStatus` implementation.
```

Comment:
@wtrocki do we need to keep this section? Should we remove it?

Reply:
This is valuable info. Not sure when // was added.

Reply:
I added it because I don't know what this section is for. The information in this section is already mentioned in the sections above. If we want to keep it, then I think we need to give a bit more context, especially about the `NetworkStatus` implementation.

Reply:
I would not been keen to overload the docs with too technical/interface information. 
All client stuff is exposed thru config options. We put efforts to add documentation to the config options and actual interfaces used there. We going to generate API docs on release. Putting more documentation here that is specific to interfaces may be just overloading things with concepts that may quickly outdated.

I think that I know what you are saying - Documentation in README.md were deeply technical and was missing some kind of top level overview on how things work from really ground level. With that in mind I think things like NetworkStatus can go away or be mentioned in more technical context bellow the document and we can extract some general info about what is being used. WDYT?

Reply:
ok, sounds good to me. Should I just remove this section then?

Reply:
Yes.